packages/main/src/Menu.ts

```diff
@@ -263,22 +266,68 @@ class Menu extends UI5Element {
 		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
+	get _list() {
+		return this.shadowRoot!.querySelector<List>("[ui5-list]")!;
+	}
+
+	get _menuItemGroups() {
+		return this.items.filter((item) : item is MenuItemGroup => item.isGroup);
+	}
+
 	get _menuItems() {
-		return this.items.filter((item): item is MenuItem => !item.isSeparator);
+		return this.items.filter((item) : item is MenuItem => !item.isSeparator && !item.isGroup);
+	}
+
+	get _menuItemsAll() {
+		const items: Array<MenuItem> = [];
+
+		this.items.forEach(item => {
+			if (item.isGroup) {
+				items.push(...(item as MenuItemGroup)._menuItems);
```

Comment:
For me filtration of menu items should happen here since it doesn't know this context and simply items property to be used.

Reply:
Group also can contain separators, which are not regular itrems, that's why I wnt to be safe and use MenuItemGroup's _menuItems getter which returns only regular items